### ISSUE TITLE
[QUESTION] JPEG decoding and encoding 3x slower than libjpeg-turbo?

### ISSUE THREAD

**Issue by Ziriax** [mono/SkiaSharp#1693]
I'm running a very simple benchmark with SkiaSharp, see code below. It basically just decodes and encodes a JPEG from and into memory.

Running this in a loop in WSL2 or Windows gives the following times:

```
Decoding took 308ms
Decoding + encoding took 738ms. Size=4198568
```

However, running libjpeg-turbo  from the command line gives

```
peter@STS-RYZEN:~/photos$ time djpeg ~/photo.jpg >/dev/null
real    162ms
```

```
peter@STS-RYZEN:~/photos$ time (djpeg ~/photo.jpg | cjpeg -q 90 >/dev/null)
real    245ms
```

So encoding + decoding is **3 times** slower in SkiaSharp?

It tried encoding and decoding to files, but that doesn't change the factor.

How is this possible? Because Skia uses libjpeg-turbo internally no? 

I'm going to try to debug this, but any help is welcome.

```cs
using System;
using System.Diagnostics;
using System.IO;
using SkiaSharp;

namespace JpegEncodingBenchmark
{
	class Program
	{
		static void Main(string[] args)
		{
			var sw = new Stopwatch();

			Console.WriteLine(Environment.OSVersion);
			
			using var data = SKData.Create(@"/home/peter/photo.jpg");

			for (;;)
			{
				sw.Restart();
				
				SKBitmap bitmap = SKBitmap.Decode(data);
				Console.WriteLine($"Decoding took {sw.ElapsedMilliseconds}ms");

				var outputData = bitmap.Encode(SKEncodedImageFormat.Jpeg, 90);
				Console.WriteLine($"Decoding + encoding took {sw.ElapsedMilliseconds}ms. Size={outputData.Size}");
				
				outputData.Dispose();
				bitmap.Dispose();
			}
		}
	}
}
```

**Comment by Ziriax** [comment]
Ow... In the `cake` files I see that `skia_use_system_libjpeg_turbo=false` everywhere?

Why is that?

In [Google's Skia](https://github.com/google/skia/blob/main/third_party/libjpeg-turbo/BUILD.gn), the default seems to be 

`skia_use_system_libjpeg_turbo = is_official_build`

That explains it why SkiaSharp is so much slower of course for loading and saving JPEG files, no SIMD is used...

**Comment by Ziriax** [comment]
No, we decided to use `libjpeg-turbo` directly, because Skia is also too strict, it rejects a lot of JPEG files that can perfectly be decoded. Well actually, that is on my current sprint goal 😉

**Comment by tomloomans88** [comment]
Thnx for your answer. I'm looking to encode SKBitmaps to PNG faster (the default method is really slow) so I will try to find a fast png encoding library and use that. Good luck with the sprint 😄

**Comment by Ziriax** [comment]
I see. Now. this issue was for JPEG encoding. 

For fast encoding to PNG, I guess it's best not use compression, or a small compression factor. 

Out of interest, I did a Google search for `parallel PNG encoder`, and found this:

https://github.com/brion/mtpng
